### PR TITLE
Fix flaky test of `testEnablePartialUpdateOnActiveActiveStore` due to no retry on verifying logic.

### PR DESCRIPTION
**Pull request by haoxu07**
## Summary, imperative, start upper case, don't end with a period

Previously, `testEnablePartialUpdateOnActiveActiveStore` was flaky due to no retry on verifying logic for consumer assignment with a fixed timeout. Besides, it has a wrong assumption:  the consumer we verify will only have one partition assignment. 

## Does this PR introduce any user-facing changes?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java [hunk 1]
```diff
@@ -1175,7 +1175,8 @@ public void testRepushWithTTLWithActiveActivePartialUpdateStore() {
 
   }
 
-  @Test(timeOut = TEST_TIMEOUT_MS)
+  // TODO: remove invocation # before merging this change.
+  @Test(timeOut = TEST_TIMEOUT_MS, invocationCount = 10)
```
[the remark this answers is not in the document]
Comment (haoxu07) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java:1179]: Sure, keeping that to demonstrate the flaky behavior has already been resolved.

**Comment by sushantmane** [comment]
>Successful run with 10 invocations: https://github.com/linkedin/venice/actions/runs/10691957458/job/29639482184?pr=1151
Successful run with 10 invocations: https://github.com/linkedin/venice/actions/runs/10691957458/job/29639482504?pr=1151


@haoxu07 - The above links are taking to UTs and not E2E results

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java [hunk 2]
```diff
@@ -1283,29 +1285,31 @@ public void testEnablePartialUpdateOnActiveActiveStore() {
...
+            1);
+        verifyConsumerThreadPoolFor(
+            storeVersionTopicV1,
+            new PubSubTopicPartitionImpl(storeVersionTopicV1, 0),
+            ConsumerPoolType.NON_CURRENT_VERSION_NON_AA_WC_LEADER_POOL,
+            1);
+        // Version 2 has active-active enabled, so each server host will ingest from two data centers for leader.
+        verifyConsumerThreadPoolFor(
+            storeVersionTopicV2,
+            new PubSubTopicPartitionImpl(realTimeTopic, 0),
```
Comment (sushantmane) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java:1304]: Maybe create local vars for RT and VT topics?

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java [hunk 3]
```diff
@@ -1430,16 +1436,17 @@ private void verifyConsumerThreadPoolFor(
...
               Map<String, TopicPartitionIngestionInfo> topicPartitionIngestionInfoMap = entry.getValue();
-              Assert.assertTrue(topicPartitionIngestionInfoMap.size() <= 1);
               for (Map.Entry<String, TopicPartitionIngestionInfo> topicPartitionIngestionInfoEntry: topicPartitionIngestionInfoMap
                   .entrySet()) {
                 String topicPartitionStr = topicPartitionIngestionInfoEntry.getKey();
-                TopicPartitionIngestionInfo topicPartitionIngestionInfo = topicPartitionIngestionInfoEntry.getValue();
-                assertEquals(pubSubTopicPartition.toString(), topicPartitionStr);
-                assertTrue(topicPartitionIngestionInfo.getConsumerIdStr().contains(consumerPoolType.getStatSuffix()));
-                regionCount += 1;
+                if (pubSubTopicPartition.toString().equals(topicPartitionStr)) {
```
Comment (sushantmane) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java:1442]: This check won't work topicPartitionIngestionContexts map is empty. Should we do something like:
```java
assertFalse(topicPartitionIngestionContexts.isEmpty());
assertTrue(topicPartitionIngestionContexts.contains(pubSubTopicPartition.toString()));
```
?

Comment (haoxu07) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java:1442]: I think overall your concern is right. But this check will only work on the server host has topic partition we are interested in. As we loop through all server hosts, I would like to add counter logic for # of replicas found per region to ensure these checks passed with expected # for every region.

Comment (sushantmane) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java:1442]: >I would like to add counter logic for # of replicas found per region to ensure these checks passed with expected # for every region.

Yeah that sounds good. Thanks

**Comment by haoxu07** [comment]
> > Successful run with 10 invocations: https://github.com/linkedin/venice/actions/runs/10691957458/job/29639482184?pr=1151
> > Successful run with 10 invocations: https://github.com/linkedin/venice/actions/runs/10691957458/job/29639482504?pr=1151
> 
> @haoxu07 - The above links are taking to UTs and not E2E results

Yeah, I removed them.

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java [hunk 4]
```diff
@@ -1283,29 +1296,35 @@ public void testEnablePartialUpdateOnActiveActiveStore() {
...
+            realTimeTopicPartition,
+            ConsumerPoolType.CURRENT_VERSION_AA_WC_LEADER_POOL,
+            NUMBER_OF_CHILD_DATACENTERS,
+            1);
+        verifyConsumerThreadPoolFor(
+            storeVersionTopicV2,
+            storeVersionTopicPartitionV2,
+            ConsumerPoolType.CURRENT_VERSION_NON_AA_WC_LEADER_POOL,
+            1,
+            REPLICATION_FACTOR - 1);
```
Comment (sushantmane) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java:1326]: Why expected replicas are one less than the RF?

**Review by sushantmane — APPROVED** [review]
LGTM. Thanks, @haoxu07!

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java (hunk 4 above)
Comment (haoxu07) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java:1326]: Discussed offline. A bit confusing here.

For hybrid store, leader is rt partition 1 per colo, follower are version topic partitions, it is RF-1.

**Pull request merged**